### PR TITLE
Fix some C# lang about BUGs

### PR DESCRIPTION
**Pull request by lailongwei**
Fix in vs c# project, 'debugdir' option invalid BUG;
FIx in linux c# project, generated makefile format error BUG(All path separator is back-slash[\], not adapted system type);
Fix vs c# project generate .user file format error BUG( The platform type is "Any CPU", but in .csproject, platform type is configured to "AnyCPU";

### CODE REVIEW

File: src/actions/vstudio/vs2005_csproj_user.lua [hunk 1]
```diff
@@ -57,6 +57,7 @@
 				if #contents[cfg] > 0 then
 					p.push('<PropertyGroup %s>', m.condition(cfg))
 					p.outln(contents[cfg])
+                    m.localDebuggerWorkingDirectory(cfg)
```
Comment (samsinsane) [inline src/actions/vstudio/vs2005_csproj_user.lua]: This should be put inside the call array `m.elements.userConfigPropertyGroup` defined on line 21 of this file.

File: src/actions/vstudio/vs2005_csproj_user.lua [hunk 2]
```diff
@@ -65,7 +66,11 @@
 		end
 	end
 
-
+    function m.localDebuggerWorkingDirectory(cfg)
```
Comment (samsinsane) [inline src/actions/vstudio/vs2005_csproj_user.lua]: This function should probably be placed in a logical place, with the other user configuration property group functions, either above or below `m.localDebuggerCommandArguments` depending on the order of the call array `m.elements.userConfigPropertyGroup` mentioned in my comment above.

File: src/actions/vstudio/_vstudio.lua [hunk 3]
```diff
@@ -271,7 +271,7 @@
 
 		local arch = architecture(cfg.system, cfg.architecture)
 		if not arch then
-			arch = iif(isnative, "x86", "Any CPU")
+			arch = iif(isnative, "x86", "AnyCPU")
```
Comment (samsinsane) [inline src/actions/vstudio/_vstudio.lua]: I'm not exactly sure what the issue you're trying to resolve is, but this has introduced another bug, the configuration is now `Debug Any CPU` instead of `Debug`.

File: src/actions/make/make_csharp.lua [hunk 4]
```diff
@@ -191,16 +192,17 @@
...
+		local sep = os.is("windows") and "\\" or "/"
+
 		_p('$(RESPONSE): %s', makefile)
 		_p('\t@echo Generating response file', prj.name)
 
 		_p('ifeq (posix,$(SHELLTYPE))')
 			_x('\t$(SILENT) rm -f $(RESPONSE)')
 		_p('else')
-			_x('\t$(SILENT) if exist $(RESPONSE) del %s', path.translate(response, '\\'))
+			_x('\t$(SILENT) if exist $(RESPONSE) del %s', path.translate(response, sep))
```
Comment (samsinsane) [inline src/actions/make/make_csharp.lua]: There's a test case for this particular output located in tests/actions/make/cs/test_response.lua on line 70. You should create two versions of this function, one for Windows and one for Linux, to ensure that it's correctly emitting the OS separator.

Comment (tritao) [inline src/actions/make/make_csharp.lua]: This particular branch should only executed on Windows so why did you need to change this separator?

File: src/actions/vstudio/_vstudio.lua (hunk 3 above)
Comment (lailongwei) [inline src/actions/vstudio/_vstudio.lua]: This issus is in visual studio c# project, when i use debugdir, debugargs option to setting my debug directory and debug arguments, bug this two options are invalidate,  open .user file and .csproj file to compare, I found in .csproj file the platform is "AnyCPU", but in .user file the platform is "Any CPU", I open the project settings in visual studio, try to config a debug directory(settings->debug) and then save the project settings, and open the .user file again, I found visual studio append my settings above, and the Platform is "AnyCPU", so I guess this is a premake bug in C# project, so I trying to modify this line code, In other language type, I have not tested it.

File: src/actions/make/make_csharp.lua (hunk 4 above)
Comment (lailongwei) [inline src/actions/make/make_csharp.lua]: I think the premake code must ignore platform differences, so the <slash> and <backslash> should be hide, so modify this line code.

File: src/actions/vstudio/vs2005_csproj_user.lua (hunk 2 above)
Comment (lailongwei) [inline src/actions/vstudio/vs2005_csproj_user.lua]: En, when I modify this code, I'm in company, and I also know this fix code can not place here... I'm very sorry, Is in a hurry, for this change, I will modify the code again and request pull again,

File: src/actions/vstudio/vs2005_csproj_user.lua (hunk 1 above)
Comment (lailongwei) [inline src/actions/vstudio/vs2005_csproj_user.lua]: Ok, thanks

**Comment by samsinsane** [comment]
There hasn't been any updates to this PR in over 2 months. If you fix up the issues raised in this PR, please feel free to reopen the PR for review.
